admin/controllers/pupil-pin.js

```diff
@@ -36,11 +37,11 @@ const getGeneratePinsList = async (req, res, next) => {
   const { htmlSortDirection, arrowSortDirection } = sortingAttributesService.getAttributes(sortingOptions, sortField, sortDirection)
   // TODO: data service call should be moved to a service
   try {
-    school = await schoolDataService.findOne({ _id: req.user.School })
+    school = await schoolDataService.sqlFindOneByDfeNumber(req.user.School)
     if (!school) {
       return next(Error(`School [${req.user.school}] not found`))
     }
-    pupils = await pinGenerationService.getPupils(school._id, sortField, sortDirection)
+    pupils = await pinGenerationService.getPupils(school.dfeNumber, sortField, sortDirection)
```

Comment:
Just so that I understand we are not quering schools based on Id anymore?

Reply:
Yes, we can, but the database has changed a little for schools, as id is now a new synthetic id, rather than the dfeNumber from before.  So `dfeNumber` is what was the Cosmos `id` field.

Initially we were forced to make the mdfenumber the PK as Cosmos does not support secondary unique indexes, so it was the only way to guarantee the dfeNumber was unique.

For the database having an `id` col as PK is mandatory (for us) - I have seen some guidelines that @GuyHarwood is writing that enforces this.